iaso/api/tasks/__init__.py

```diff
@@ -70,3 +79,211 @@ def get_queryset(self):
         profile = self.request.user.iaso_profile
         order = self.request.query_params.get("order", "created_at").split(",")
         return Task.objects.filter(account=profile.account).order_by(*order)
+
+
+class ExternalTaskSerializer(TaskSerializer):
+    def update(self, task, validated_data):
+        has_progress_message = validated_data.get("progress_message", None) is not None
+        has_status = validated_data.get("status", None) is not None
+        has_progress_value = validated_data.get("progress_value", None) is not None
+        has_end_value = validated_data.get("end_value", None) is not None
+        if (has_status or has_progress_value or has_progress_message or has_end_value) and not task.external:
+            raise serializers.ValidationError({"external": "Cannot modify non external tasks"})
+        if validated_data.get("should_be_killed", None) is not None:
+            task.should_be_killed = validated_data["should_be_killed"]
+            if validated_data["should_be_killed"]:
+                task.status = KILLED
+        if has_status:
+            task.status = validated_data["status"]
+            if (
+                validated_data["status"] == SUCCESS
+                or validated_data["status"] == ERRORED
+                or validated_data["status"] == KILLED
+            ):
+                task.ended_at = datetime.now()
+        if has_progress_message:
+            task.progress_message = validated_data["progress_message"]
+        if has_progress_value:
+            task.progress_value = validated_data["progress_value"]
+        if has_end_value:
+            task.end_value = validated_data["end_value"]
+        task.save()
+        return task
+
+
+class ExternalTaskPostSerializer(serializers.Serializer):
+    slug = serializers.CharField()
+    config = data = serializers.JSONField()
+    id_field = data = serializers.JSONField()
+
+    def validate(self, attrs):
+        validated_data = super().validate(attrs)
+        request = self.context["request"]
+        slug = request.data.get("slug", None)
+        config = request.data.get("config", None)
+        id_field = request.data.get("id_field", None)
+        error = {}
+
+        if slug is None:
+            error["slug"] = "This field is mandatory"
+
+        id_field_count = len(id_field.keys())
+        if id_field_count > 1:
+            error["non_field_errors"] = f"id_field should only have one property, got {id_field_count}"
+
+        if error:
+            raise serializers.ValidationError(error)
+
+        res = {**validated_data}
+        res["slug"] = slugify(slug)
+        res["config"] = config
+        res["id_field"] = id_field
```

Comment:
I think this is not required as this data is in `validated_data` and has the same name

Reply:
In this case you don't need to unpack them at the beginning of this function ...

Reply:
It breaks the test if I remove them. I'm getting the values from the request data, not from validated_data. I think it's because the fields don't exist on the model or something